### PR TITLE
Flag to add session attributes on request

### PR DESCRIPTION
**Pull request by dnetoMOB**
Adds the `withSessionAttributes` property that allows a request to be initialized with the provided attributes object.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -449,6 +450,12 @@ module.exports = {
 						}
 						return ctx.succeed(result);
 					};
+
+					// adds values from withSessionAttributes to the session
+					if (currentItem.withSessionAttributes) {
+						var session = request.session.attributes;
+						for(var x in currentItem.withSessionAttributes) session[x] = currentItem.withSessionAttributes[x];
```
[the remark this answers is not in the document]
Comment (hoegertn) [reply index.js]: Additionally, please change the following:

- use expanded form of `for`
- check if session has own property x
- rename `x` to a meaningful name

**Comment by dnetoMOB** [comment]
- For loop expand to multi-line expression.
- renamed loop variable to more descriptive name
- added check so that an attribute provided in `withSessionAttributes` should not be added to the session if it already exists.

File: index.js [hunk 2]
```diff
@@ -449,6 +450,17 @@ module.exports = {
 						}
 						return ctx.succeed(result);
 					};
+
+					// adds values from withSessionAttributes to the session
+					if (currentItem.withSessionAttributes) {
+						var session = request.session.attributes;
+
+						for (var newAttribute in currentItem.withSessionAttributes) {
+							if (typeof session[`${newAttribute}`] === "undefined") {
```
Comment (hoegertn) [inline index.js]: Why not `if (!session[newAttribute]) {`

File: index.js [hunk 3]
```diff
@@ -449,6 +450,17 @@ module.exports = {
...
+								session[`${newAttribute}`] = currentItem.withSessionAttributes[`${newAttribute}`];
```
Comment (hoegertn) [inline index.js]: Why do you use template strings here?

Comment (dnetoMOB) [reply index.js]: Template strings removed (it was an odd suggestion from a friend).

File: index.js (hunk 2 above)
Comment (dnetoMOB) [reply index.js]: Good suggestion. I've had issues before with `"undefined"` vs `undefined`, but that syntax should clear things up.

**Pull request merged**